api/types/types.go

```diff
@@ -396,20 +396,38 @@ type NetworkResource struct {
 	Name       string                         // Name is the requested name of the network
 	ID         string                         `json:"Id"` // ID uniquely identifies a network on a single machine
 	Created    time.Time                      // Created is the time the network created
-	Scope      string                         // Scope describes the level at which the network exists (e.g. `global` for cluster-wide or `local` for machine level)
+	Scope      string                         // Scope describes the level at which the network exists (e.g. `swarm` for cluster-wide or `local` for machine level)
 	Driver     string                         // Driver is the Driver name used to create the network (e.g. `bridge`, `overlay`)
 	EnableIPv6 bool                           // EnableIPv6 represents whether to enable IPv6
 	IPAM       network.IPAM                   // IPAM is the network's IP Address Management
 	Internal   bool                           // Internal represents if the network is used internal only
 	Attachable bool                           // Attachable represents if the global scope is manually attachable by regular containers from workers in swarm mode.
 	Ingress    bool                           // Ingress indicates the network is providing the routing-mesh for the swarm cluster.
+	References []NetworkReference             // References points to a list of references for this network.
```

Comment:
Yes, the first you said.
Reason we opted for the reference construct is because from comments that the `ConfigFrom` seemed more appropriate for a flag rather than for carrying the name of the network which provides the configuration.

Also to provide this via a more generic structure, so that if in future this  functionality can be provided via other means, we only need to deprecate the operation mode at runtime and no deprecation of a API field is needed.